crates/fuel-core/src/query/blob.rs

```diff
@@ -0,0 +1,38 @@
+use crate::graphql_api::ports::{
+    OffChainDatabase,
+    OnChainDatabase,
+};
+use fuel_core_storage::{
+    not_found,
+    tables::BlobData,
+    Result as StorageResult,
+    StorageAsRef,
+};
+use fuel_core_types::fuel_tx::BlobId;
+
+pub trait BlobQueryData: Send + Sync {
+    fn blob_id(&self, id: BlobId) -> StorageResult<BlobId>;
+    fn blob_bytecode(&self, id: BlobId) -> StorageResult<Vec<u8>>;
+}
+
+impl<D: OnChainDatabase + OffChainDatabase + ?Sized> BlobQueryData for D {
+    fn blob_id(&self, id: BlobId) -> StorageResult<BlobId> {
+        let blob_exists = self.storage::<BlobData>().contains_key(&id)?;
+
+        if blob_exists {
+            Ok(id)
+        } else {
+            Err(not_found!(BlobData))
+        }
+    }
```

Comment:
Feels a bit weird to have an identity function endpoint that tests inside if the blob exists. Typically things like this would return a boolean.

Reply:
I agree. I did it this way to be consistent with the implementation in `crates/fuel-core/src/query/contract.rs`. I'm happy to change this to be an `blob_id_exists` method that returns a boolean, but then I would also change any other methods using this pattern including the `contract_id` method in `crates/fuel-core/src/query/contract.rs`, which would add some refactor-noise to this PR.

Do you prefer if I do this now or as a follow-up?

Reply:
Hmmm I didn't realize the `contract_id` was like this. I'm feeling the backwards compatibility is more important than consistency here.

Reply:
Offline discussion: backwards compatibility here is only relevant if the `contract_id` method would have been part of a public endpoint, which it is not - so I will do the refactor.